Tella/Data/Database/UwaziDatabase.swift

```diff
@@ -62,6 +70,28 @@ extension TellaDataBase: UwaziTemplateProtocol {
             return nil
         }
     }
+    
+    func updateUwaziTemplate(template: CollectedTemplate) -> Int? {
+        do {
+            let valuesToUpdate = [
```

Comment:
In my opinion it's better to use the `encode/decode` fuctions to prepare the model and then convert it to `dictionnary` 
after that convert it to  `KeyValue` array 
there is an example in `UwaziEntityInstance` class  https://github.com/Horizontal-org/Tella-iOS/pull/142/files#diff-bbe17c95214006c21c85223ce5878227d41e2a1734377231206bd283c0ad1178  and the implementation in ` func addUwaziEntityInstance(entityInstance : UwaziEntityInstance) -> Result<Int, Error>` function in [Tella/Data/Database/UwaziDatabase.swift ](https://github.com/Horizontal-org/Tella-iOS/pull/142/files#diff-b59519d7aeca9338a5aae62f11c0ccd5ac0c46226391ec66e5ba7faff45502ed) class